_episodes/12-large-files.md

```diff
@@ -3,14 +3,164 @@ title: "Large Files"
+
+~~~
+Tracking "report.pdf"
+~~~
+{: .output}
+
+Having done so, we can inspect the repository and we learn that a new file `.gitattributes`
+was added to the repository. 
+
+~~~
+git status
+~~~
+{: .language-bash}
+
+~~~
+On branch main
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+	.gitattributes
+	report.pdf
+~~~
+{: .output}
+
+~~~
+cat .gitattributes
+~~~
+{: .language-bash}
+
+~~~
+report.pdf filter=lfs diff=lfs merge=lfs -text
+~~~
+{: .output}
+
+Similar to `.gitignore` this file is part of the repository
+itself in order to share it with all your collaborators on this project.
+We therefore craft a commit that contains it:
+
+~~~
+git add .gitattributes
+git commit -m "Setup LFS tracking"
+~~~
+{: .language-bash}
+
+Now, we are ready to add the large file to the repository the same way we would with any other file:
+
+~~~
+git add report.pdf
+git commit -m "Add final report to the repository"
+~~~
+{: .language-bash}
+
+Pushing our commits to the remote repository, we can see in the console
+output, that our LFS data was transferred to the remote server separately.
+
+~~~
+git push origin main
+~~~
+{: .language-bash}
+
+~~~
+Uploading LFS objects: 100% (1/1), 17 B | 0 B/s, done.                          
+~~~
+{: .output}
+
+~~~
+LFS tracking is not limited to explicitly spelled out filenames. Instead, wildcard
+patterns can be passed to `git lfs track`. However, you should be careful to quote
+these patterns, as they might otherwise get expanded by to existing files by your shell.
+For example, tracking all PDFs with LFS could be achieved with the following command:
+
+~~~
+git lfs track "*.pdf"
+~~~
+{: .language-bash}
+~~~
+{: .callout}
+
+~~~
+Although `git lfs` by design solves the problem of storing large files in Git
+repositories, there are some practical hurdles that you should consider before
+introducing LFS into your project:
+
+* The `git lfs` command is a separately maintained extension to the Git core. It is
+  therefore not part of most Git distributions, but needs to be installed separately.
+  Using it in your project will require you to educate your users about LFS and how
+  to install it. Depending on your target audience, you should carefully consider
+  whether the benefits outweigh this disadvantage.
+* Users that do not have `git lfs` installed will not be notified by Git. They
+  will see the files, but the content will be Git metadata instead of the actual content.
+  Trying to work with those files will typically produce cryptic error messages.
+* Some hosting providers - most notably GitHub - apply restrictive quotas to LFS storage.
+  On the free plan, GitHub currently allows 1GB of storage and 1 GB bandwidth per month.
+  As the band width quota counts every single clone by users, **LFS should currently
+  be considered unusable on the GitHub free plan.**
+~~~
+{: .caution}
+
+TODO: Write about `git annex`
+
+TODO: Write about other potential solutions.
```

Comment:
Can you please add the ideas from the slide so we can keep track of them:
TODO: Write about other potential solutions, including ignore them in Git repo, handle versioning somewhere else. 

To completely remove a file from history (for the desperate only):
- Squash out all the commits that contains that file
- git filter-branch
- [BFG repo cleaner](https://rtyley.github.io/bfg-repo-cleaner/)
- Start afresh in a new repo

Reply:
Added to #28 (to avoid TODO notes in the repository)